libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -632,6 +632,30 @@ int AP_HAL__I2CDevice_read_registers(lua_State *L) {
     return success;
 }
 
+int AP_HAL__UARTDriver_readstring(lua_State *L) {
+    binding_argcheck(L, 2);
+
+    AP_HAL::UARTDriver * ud = *check_AP_HAL__UARTDriver(L, 1);
+
+    const uint16_t count = get_uint16_t(L, 2);
+    uint8_t *data = (uint8_t*)malloc(count);
+    if (data == nullptr) {
+        return 0;
+    }
+
+    const auto ret = ud->read(data, count);
```

Comment:
I wonder if it would be useful to return the read length as a second value. It would save a having to do a string length in the script.

Reply:
That is to say, people should not assume that they got the length they asked for. Maybe it would be enough to just mention that that might happen in the description.

Reply:
I've added it to the docs